tests/sqlparser_mssql.rs

```diff
@@ -1024,6 +1024,17 @@ fn parse_create_table_with_identity_column() {
     }
 }
 
+#[test]
```

Comment:
Ah sorry I forgot to mention this part earlier, the test looks reasonable but now with the method on the dialect, can we move the test to common.rs and write it in a [similar style to this](https://github.com/apache/datafusion-sqlparser-rs/blob/cb7eeed2aa7b4fdeb402e0df6e56a8b662633efb/tests/sqlparser_common.rs#L2781)? that would allow any future dialects with the same support to be covered automatically by the test.

Also for this feature, can we include an assertion that for the dialects that don't support the flag, verifying that they indeed end up with an unnamed select item?